### PR TITLE
Add Automation to RoveComm Manifest Updates

### PR DESCRIPTION
**Pull request by Byrdman32**
### Automating Submodule Updates for RoveComm in RoveComm_CPP

This GitHub Action automates the process of keeping the `RoveComm_Base` submodule in the `RoveComm_CPP` repository up to date with its latest changes. The workflow is triggered by external dispatch events and automatically performs the following steps:

1. **Checks Out the Main Repository**: 
   - Ensures the main repository is checked out with all submodules initialized.
   
2. **Fetches and Updates the RoveComm_Base Submodule**: 
   - The action fetches the latest changes from the `RoveComm_Base` repository and updates the submodule to point to the latest commit.

3. **Creates a New Branch with a Timestamp**:
   - If the submodule has changes, the action creates a new branch named `sync/rovecomm-YYYYMMDD-HHMMSS`, which includes a timestamp to ensure branch uniqueness.

4. **Pushes the Branch and Opens a Pull Request**:
   - The new branch is pushed to the repository, and a pull request is automatically created with details about the submodule update. This pull request is ready for review and merge.

### Benefits of the Workflow:
- **Automated Updates**: You no longer need to manually check and update the submodule reference in the repository. This GitHub Action ensures that the submodule is always up to date.
- **Unique Branch Naming**: The use of a timestamp in the branch name prevents conflicts and ensures that each update is easily identifiable.
- **Effortless Pull Requests**: By automatically generating pull requests for each update, the team can quickly review and merge changes without manual intervention.

---

### Example PR Created by the Action:
Check out [this pull request](https://github.com/byrdman32/test/pull/1) as an example of a pull request created by the action, which updates the RoveComm submodule to the latest commit.

---

### Usage:
1. **Trigger the Workflow**: The workflow is triggered using repository dispatch events, making it easy to integrate with other automation or manually trigger updates when needed.
2. **Review and Merge the Pull Request**: Once the pull request is created, simply review the submodule changes and merge them into the main branch.

---

This automation will help ensure that the `RoveComm_CPP` repository always stays in sync with the latest changes from the `RoveComm` repository, reducing manual work and potential errors.

### CODE REVIEW

**Review by ClayJay3 — APPROVED** [review]

**Comment by Byrdman32** [comment]
I'm going to hold off on merging this until I get it to work correctly with a webhook so that I am not just running a GitHub action once every minute forever.

File: .github/workflows/sync-rovecomm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          git submodule update --remote --merge data/RoveComm
+          cd data/RoveComm
+          git checkout $(git rev-parse --abbrev-ref HEAD)
+          cd -
+
+      - name: Commit changes if RoveComm was updated
+        id: commit_changes  # Set an ID so we can refer to this step's output
+        run: |
+          git add data/RoveComm
+          if git diff-index --quiet HEAD; then
```
Comment (ClayJay3) [inline .github/workflows/sync-rovecomm]: Will git diff work how you're expecting if the changes are staged but not committed?

File: .github/workflows/sync-rovecomm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            echo "No changes to commit"
+            echo "::set-output name=changes_made::false"
```
Comment (ClayJay3) [inline .github/workflows/sync-rovecomm]: According to GitHub set output is deprecated? 

Maybe change to using this `echo "changes_made=true" >> $GITHUB_ENV`

File: .github/workflows/sync-rovecomm (hunk 1 above)
Comment (Byrdman32) [reply .github/workflows/sync-rovecomm]: Resolved by changing how this action works.

File: .github/workflows/sync-rovecomm (hunk 2 above)
Comment (Byrdman32) [reply .github/workflows/sync-rovecomm]: These are no longer needed.

**Review by ClayJay3 — APPROVED** [review]

**Pull request merged**